composeApp/src/commonMain/kotlin/org/ooni/probe/ui/settings/proxy/ProxyScreen.kt

```diff
@@ -0,0 +1,170 @@
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom_Port
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Custom_Protocol
+import ooniprobe.composeapp.generated.resources.Settings_Proxy_Enabled
+import ooniprobe.composeapp.generated.resources.back
+import org.jetbrains.compose.resources.stringResource
+import org.ooni.probe.data.models.ProxyProtocol
+import org.ooni.probe.data.models.ProxyType
+
+@Composable
+fun ProxyScreen(
+    onEvent: (ProxyViewModel.Event) -> Unit,
+    state: ProxyViewModel.State,
+) {
+    Column {
+        TopAppBar(
+            title = {
+                Text(stringResource(Res.string.Settings_Proxy_Enabled))
+            },
+            navigationIcon = {
+                IconButton(onClick = {
+                    onEvent(ProxyViewModel.Event.BackClicked)
+                }) {
+                    Icon(
+                        Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = stringResource(Res.string.back),
+                    )
+                }
+            },
+        )
+        Column(Modifier.selectableGroup()) {
+            ProxyType.entries.forEach { text ->
+                Row(
+                    Modifier.fillMaxWidth().height(56.dp).selectable(
+                        selected = (text == state.proxyType),
+                        onClick = {
+                            onEvent(
+                                ProxyViewModel.Event.ProtocolTypeSelected(
+                                    protocol = text,
+                                ),
+                            )
+                        },
+                        role = Role.RadioButton,
+                    ).padding(horizontal = 16.dp),
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    RadioButton(
+                        selected = (text == state.proxyType),
+                        onClick = null,
+                    )
+                    Text(
+                        text = stringResource(text.label),
+                        style = MaterialTheme.typography.bodyLarge,
+                        modifier = Modifier.padding(start = 16.dp),
+                    )
+                }
+            }
+        }
+        if (state.proxyType == ProxyType.CUSTOM) {
+            Row(
+                modifier = Modifier.fillMaxWidth().padding(8.dp),
+                horizontalArrangement = Arrangement.spacedBy(8.dp),
+            ) {
+                var expanded by remember { mutableStateOf(false) }
+                var hostname by rememberSaveable { mutableStateOf(state.proxyHost ?: "") }
+                var port by rememberSaveable { mutableStateOf(state.proxyPort?.toString() ?: "") }
+                val protocols =
+                    listOf(ProxyProtocol.HTTP, ProxyProtocol.HTTPS, ProxyProtocol.SOCKS5)
+                OutlinedTextField(
```

Comment:
Tapping on this field still doesn't open the dropdown menu. You need to tap on the dropdown arrow.